lib/mix/lib/mix/tasks/compile.elixir.ex

```diff
@@ -127,15 +127,22 @@ defmodule Mix.Tasks.Compile.Elixir do
 
     with_logger_app(project, fn ->
       Mix.Project.with_build_lock(project, fn ->
-        Mix.Compilers.Elixir.compile(
-          manifest,
-          srcs,
-          dest,
-          cache_key,
-          Mix.Tasks.Compile.Erlang.manifests(),
-          Mix.Tasks.Compile.Erlang.modules(),
-          opts
-        )
+        {status, warnings, modules_diff} =
+          Mix.Compilers.Elixir.compile(
+            manifest,
+            srcs,
+            dest,
+            cache_key,
+            Mix.Tasks.Compile.Erlang.manifests(),
+            Mix.Tasks.Compile.Erlang.modules(),
+            opts
+          )
+
+        if modules_diff do
+          Mix.Task.Compiler.notify_modules_compiled(modules_diff)
```

Comment:
I was going to suggest moving this call inside `Mix.Compilers.Elixir` but, to be honest, it is all the same. We will actually know which way is better once we do the Erlang one, because that may force us to go one way or the other. So let's keep this as is, but we may be forced to move it inside `Mix.Compilers.*` in a follow up PR.

Reply:
For both compilers we want to send the same event, so it's effectively more generic no?

Reply:
Yes, it is the same event, but I believe Mix.Compilers.Erlang may be used by other projects, so we cannot change its return type. :(

Reply:
Ohh I see, I though you mean where `notify_modules_compiled` is defined, but it's about where we call it. I though here is a better place and both `Mix.Compilers.{Elixir,Erlang}` are marked as private modules, but if they are effectively public, then yeah, we can make the broadcast there also.

Reply:
The Erlang one, last time I checked, was used by third party tools. The Elixir one isn't. This is why the Erlang one will be deciding factor (for consistency), but we can do it in another PR. :)